include/emp/Evolve/Systematics.hpp

```diff
@@ -545,100 +602,270 @@ namespace emp {
+      //             std::ranges::ref_view(outside_taxa)};
+      // for (emp::Ptr<taxon_t> tax : all | std::views::join) {
+      for (emp::Ptr<taxon_t> tax : active_taxa) {
+        result.push_back(fun(tax));
+      }
+      for (emp::Ptr<taxon_t> tax : ancestor_taxa) {
+        result.push_back(fun(tax));
+      }
+      for (emp::Ptr<taxon_t> tax : ancestor_taxa) {
+        result.push_back(fun(tax));
+      }
+
+      return result;
+    }
+
+    // ===== Functions for querying phylogeny/systematics manager internal state ====
+
+    // Currently using raw pointer because of a weird bug in emp::Ptr. Should switch when fixed.
     std::unordered_set< Ptr<taxon_t>, hash_t > * GetActivePtr() { return &active_taxa; }
+    /// @returns set of active (extant/living) taxa0
     const std::unordered_set< Ptr<taxon_t>, hash_t > & GetActive() const { return active_taxa; }
+    /// @returns set of ancestor taxa (extinct, but have active descendants)
     const std::unordered_set< Ptr<taxon_t>, hash_t > & GetAncestors() const { return ancestor_taxa; }
+    /// @returns set of outside taxa (extinct, with no active descendants)
     const std::unordered_set< Ptr<taxon_t>, hash_t > & GetOutside() const { return outside_taxa; }
 
-    /// How many taxa are still active in the population?
+    /// @returns the number of taxa that are still active in the population
     size_t GetNumActive() const { return active_taxa.size(); }
 
-    /// How many taxa are ancestors of living organisms (but have died out themselves)?
+    /// @returns the number of taxa that are ancestors of living organisms (but have died out themselves)
     size_t GetNumAncestors() const { return ancestor_taxa.size(); }
 
-    /// How many taxa are stored that have died out, as have their descendents?
+    /// @returns the number of taxa that are stored that have died out, as have their descendents
     size_t GetNumOutside() const { return outside_taxa.size(); }
 
-    /// How many taxa are in the current phylogeny?
+    /// @returns the number of taxa that are in the current phylogeny
     size_t GetTreeSize() const { return GetNumActive() + GetNumAncestors(); }
 
-    /// How many taxa are stored in total?
+    /// @returns the number of taxa that are stored in total
     size_t GetNumTaxa() const { return GetTreeSize() + GetNumOutside(); }
 
-    int GetMaxDepth() {
-      if (max_depth != -1) {
-        return max_depth;
-      }
+    /// @returns the phylogenetic depth (lineage length) of the taxon with
+    /// the longest lineage out of all active taxa
+    int GetMaxDepth();
 
-      for (auto tax : active_taxa) {
-        int depth = tax->GetDepth();
-        if (depth > max_depth) {
-          max_depth = depth;
-        }
-      }
-      return max_depth;
+    /// @returns the taxon that will be used as the parent
+    /// of the next taxon created via the version of AddOrg
+    /// that does not accept a parent
+    Ptr<taxon_t> GetNextParent() {
```

Comment:
probably because this is returning a pointer having it non-const as it is makes sense?

Reply:
there's a few other functions with this pattern that might get const'ed if decided the other way

Reply:
Yeah, I think it's reasonable for these all to be const. Fundamentally the taxon objects are different objects than the systematics manager, so I think it's okay to return pointers to them from const methods